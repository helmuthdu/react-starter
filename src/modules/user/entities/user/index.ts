export * from './user.type.ts';
export * from './user.model';
