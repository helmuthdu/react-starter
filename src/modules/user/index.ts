export { routes } from './routes';
export * from './stores';
