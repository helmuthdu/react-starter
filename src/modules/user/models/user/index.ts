export * from './user.scheme';
export * from './user.model';
