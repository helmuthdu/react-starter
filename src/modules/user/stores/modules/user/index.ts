export * from './actions';
export * from './selectors';
export * from './reducer';
export * from './state';
export * from './types';

export const name = 'user';
