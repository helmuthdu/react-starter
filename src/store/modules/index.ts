import * as loading from './loading';

export { loading };
