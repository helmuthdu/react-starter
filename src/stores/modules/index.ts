import * as localeStore from './locale';
import * as notificationStore from './notification';

export type State = {
  locale: localeStore.State;
  notification: notificationStore.State;
};

export type Action = localeStore.Action | notificationStore.Action;

export type Reducer = localeStore.Reducer | notificationStore.Reducer;

export const reducer: Reducer = {
  ...localeStore.reducer,
  ...notificationStore.reducer
};

export const initialState: State = {
  locale: localeStore.initialState,
  notification: notificationStore.initialState
};
