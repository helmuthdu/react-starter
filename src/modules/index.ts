import { routes as rootRoutes } from './root';
import * as userModule from './user';

export type State = userModule.State;

export type Action = userModule.Action;

export type Mutations = userModule.Mutations;

export const routes = [...rootRoutes, ...userModule.routes];

export const stores = { ...userModule.stores };

export default {
  routes,
  stores
};
