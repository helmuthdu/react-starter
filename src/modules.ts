import * as authModule from './modules/auth';
import * as mainModule from './modules/main';
import * as appStores from './store/modules';

export type AppState = authModule.State & {
  loading: appStores.loading.State;
};

export const routes = [...mainModule.routes, ...authModule.routes];
export const stores = [...authModule.stores, ...Object.values(appStores)];

export default {
  routes,
  stores
};
