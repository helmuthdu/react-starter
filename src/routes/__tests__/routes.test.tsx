import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { AppRoutes } from '..';
import { initialState } from '../../modules/user/stores/modules/user';

describe('App router', () => {
  const middlewares = [thunk];

  const store = configureMockStore(middlewares)({ user: initialState });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <AppRoutes routes={[]} />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
