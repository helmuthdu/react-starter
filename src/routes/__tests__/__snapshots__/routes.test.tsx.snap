// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App router should match snapshot 1`] = `<DocumentFragment />`;
